drivers/pinctrl/pinctrl_smartbond.c

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (c) 2022, Renesas Electronics Corporation
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/drivers/pinctrl.h>
+#include <DA1469xAB.h>
+
+/** Utility macro to retrieve starting mode register and pin count for GPIO port from DT */
+#define GPIO_PORT_ENTRY(nodelabel)							\
+		{ DT_REG_ADDR_BY_IDX(DT_NODELABEL(nodelabel), 1),			\
+		  DT_REG_SIZE_BY_IDX(DT_NODELABEL(nodelabel), 1) / sizeof(uint32_t) }
+
+struct gpio_port {
+    uint32_t p0_mode_addr;
+    uint8_t pin_count;
+};
+
+static const struct gpio_port smartbond_gpio_ports[] = {
+	GPIO_PORT_ENTRY(gpio0),
+	GPIO_PORT_ENTRY(gpio1),
+};
```

Comment:
please add gpio driver first, this is not bisectable as these nodes don't exist yet